ql/src/java/org/apache/hadoop/hive/ql/HookRunner.java

```diff
@@ -39,57 +36,27 @@
 import org.apache.hadoop.hive.ql.parse.HiveSemanticAnalyzerHook;
 import org.apache.hadoop.hive.ql.parse.HiveSemanticAnalyzerHookContext;
 import org.apache.hadoop.hive.ql.session.SessionState;
-import org.apache.hadoop.hive.ql.session.SessionState.LogHelper;
 import org.apache.hive.common.util.HiveStringUtils;
 
+import static org.apache.hadoop.hive.ql.hooks.HookContext.HookType.*;
+
 /**
  * Handles hook executions for {@link Driver}.
  */
 public class HookRunner {
 
   private static final String CLASS_NAME = Driver.class.getName();
   private final HiveConf conf;
-  private LogHelper console;
-  private List<QueryLifeTimeHook> queryHooks = new ArrayList<>();
-  private List<HiveSemanticAnalyzerHook> saHooks = new ArrayList<>();
-  private List<HiveDriverRunHook> driverRunHooks = new ArrayList<>();
-  private List<ExecuteWithHookContext> preExecHooks = new ArrayList<>();
-  private List<ExecuteWithHookContext> postExecHooks = new ArrayList<>();
-  private List<ExecuteWithHookContext> onFailureHooks = new ArrayList<>();
-  private boolean initialized = false;
+  private final HooksLoader loader;
```

Comment:
this is great!
since from now on we can also dynamically add new hooks to it at runtime - we may rename it from "Loader" to something else.

Reply:
Rename it to HiveHooks instead.